src/web-animations-next-animation.js

```diff
@@ -25,13 +27,48 @@
     this._animation = null;
     this._childAnimations = [];
     this._callback = null;
+    this._oldPlayState = 'idle';
     this._rebuildUnderlyingAnimation();
     // Animations are constructed in the idle state.
     this._animation.cancel();
+    this._updatePromises();
   };
 
   scope.Animation.prototype = {
+    _updatePromises: function() {
+      var oldPlayState = this._oldPlayState;
+      var newPlayState = this.playState;
+      if (this._readyPromise && newPlayState !== oldPlayState) {
+        if (newPlayState == 'idle') {
+          if (this._readyPromiseState == 'pending') {
```

Comment:
> It's worth noting that the guard here is sort of redundant in that you can't reject or resolve an already completed promise (i.e., if _readyPromiseState isn't 'pending'). The promise just ignores you (this is by spec).
> You might be able to simplify the _{ready,reject}PromiseState checks this way.

True. I copied these checks from Blink. I kind of like that the code as-is explains that you only reject when moving from pending to idle, but I think you are right that we should remove the check.